java/src/main/native/src/CuFileJni.cpp

```diff
@@ -306,14 +307,167 @@ JNIEXPORT void JNICALL Java_ai_rapids_cudf_CuFile_destroyDriver(JNIEnv *env, jcl
   CATCH_STD(env, );
 }
 
+/**
+ * @brief Create a new cuFile buffer wrapper.
+ *
+ * @param env The JNI environment.
+ * @param device_pointer Pointer address to the device buffer.
+ * @param size The size of the device buffer.
+ * @param register_buffer If true, register the cuFile buffer.
+ * @return Pointer address to the new buffer wrapper instance.
+ */
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_CuFileBuffer_create(JNIEnv *env, jclass,
+                                                                jlong device_pointer, jlong size,
+                                                                jboolean register_buffer) {
+  try {
+    cudf::jni::auto_set_device(env);
+    auto *buffer =
+        new cufile_buffer(reinterpret_cast<void *>(device_pointer), size, register_buffer);
+    return reinterpret_cast<jlong>(buffer);
+  }
+  CATCH_STD(env, 0);
+}
+
+/**
+ * @brief Destroy the given cuFile buffer wrapper.
+ *
+ * @param env The JNI environment.
+ * @param pointer Pointer address to the buffer wrapper instance.
+ */
+JNIEXPORT void JNICALL Java_ai_rapids_cudf_CuFileBuffer_destroy(JNIEnv *env, jclass,
+                                                                jlong pointer) {
+  try {
+    cudf::jni::auto_set_device(env);
+    if (pointer != 0) {
```

Comment:
Nit: a delete of a nullptr is a no-op, so this check is redundant.

Reply:
Done.